### PR TITLE
Translate ruby 3.0 release (zh_cn)

### PR DESCRIPTION
**Pull request by dsh0416**

### CODE REVIEW

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 1]
```diff
@@ -0,0 +1,388 @@
+---
+layout: news_post
+title: "Ruby 3.0.0 现已发布"
+author: "naruse"
+translator: Delton Ding
+date: 2020-12-25 00:00:00 +0000
+lang: zh_cn
+---
+
+我们很高兴宣布 Ruby {{ release.version }} 已发布。自 2015 年起，我们就开始努力开发 Ruby 3，其目标在更高的性能、并发性和更安全的类型。尤其是性能上，Matz 表示「Ruby 3 会比 Ruby 2 快 3 倍」，也就是 [Ruby 3x3](https://blog.heroku.com/ruby-3-by-3)。
```
Comment (stomar) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:10]:
```suggestion
我们很高兴宣布 Ruby 3.0.0 已发布。自 2015 年起，我们就开始努力开发 Ruby 3，其目标在更高的性能、并发性和更安全的类型。尤其是性能上，Matz 表示「Ruby 3 会比 Ruby 2 快 3 倍」，也就是 [Ruby 3x3](https://blog.heroku.com/ruby-3-by-3)。
```

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 2]
```diff
@@ -0,0 +1,388 @@
...
+
+{% assign release = site.data.releases | where: "version", "3.0.0" | first %}
+
+<img src='https://cache.ruby-lang.org/pub/media/ruby3x3.png' alt='Optcarrot 3000 frames' width='100%' />
+
+在 [Optcarrot 基准测试中](https://github.com/mame/optcarrot) 我们测量了单线程 NES 游戏模拟器的工作负载下的性能，其比 Ruby 2.0 快了 3 倍！ <details>该结果是 Ruby 3.0 在下面的环境中测试得到的 https://benchmark-driver.github.io/hardware.html. [8c510e4095](http://github.com/ruby/ruby/commit/8c510e4095)。取决于你所运行的环境和基准测试的类型，可能提升不足 3 倍。</details>
```
Comment (stomar) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:16]:
```suggestion
在 [Optcarrot 基准测试中](https://github.com/mame/optcarrot) 我们测量了单线程 NES 游戏模拟器的工作负载下的性能，其比 Ruby 2.0 快了 3 倍！ <details>该结果是 Ruby 3.0 在下面的环境中测试得到的 [benchmark-driver.github.io/hardware.html](https://benchmark-driver.github.io/hardware.html). [8c510e4095](http://github.com/ruby/ruby/commit/8c510e4095)。取决于你所运行的环境和基准测试的类型，可能提升不足 3 倍。</details>
```

**Comment by dsh0416** [comment]
Fixed. Thank you for reviewing. @stomar

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 3]
```diff
@@ -0,0 +1,388 @@
...
+
+MJIT 引入了大量的提升。详见 NEWS。
+
+对于 Ruby 3.0 来说，JIT 将能在某些特定的工作负载下得到性能提升，例如游戏（[Optcarrot](https://benchmark-driver.github.io/benchmarks/optcarrot/commits.html#chart-1)）、AI（[Rubykon](https://benchmark-driver.github.io/benchmarks/rubykon/commits.html)），以及其它大多数时间都在反复调用某些特定方法的应用中。
+
+虽然 Ruby 3.0 [大幅降低了 JIT 后代码的大小](https://twitter.com/k0kubun/status/1256142302608650244)，对于例如 Rails 这样的负载还没有做好准备。其调用大量不同的方法，在使用 JIT 后会遇到 CPU 指令缓存命中率降低的问题。Ruby 3.1 会在这一问题上进行进一步优化提升，敬请关注。
+
+## 并发/并行
+
+> 现在是多核的时代，并发性很重要。有了 Ractor 和异步 Fiber，Ruby 将会成为一门真正的并发语言。—— Marz
```
Comment (hlcfan) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:44]: Matz 😅

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 4]
```diff
@@ -0,0 +1,388 @@
...
+$ typeprof test.rb
+# Classes
+class User
+  attr_reader name : String
+  attr_reader age : Integer
+  def initialize : (name: String, age: Integer) -> [String, Integer]
+end
+```
+
+You can run TypeProf by saving the input as "test.rb" and invoke a command called "typeprof test.rb".
```
Comment (hlcfan) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:190]: Please remove this line.

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 5]
```diff
@@ -0,0 +1,388 @@
...
+
+> 当我第一次在会议上提到「Ruby 3x3」的时候，大多数核心团队的成员都觉得 Matz 在吹牛。事实上，我也这么觉得。但是我们做到了。我很高兴看到核心开发团队完成了 Ruby 3.0 比 2.0 快 3 倍的目标（在一些基准测试中）。—— Matz
+
+### MJIT
+
+MJIT 引入了大量的提升。详见 NEWS。
+
+对于 Ruby 3.0 来说，JIT 将能在某些特定的工作负载下得到性能提升，例如游戏（[Optcarrot](https://benchmark-driver.github.io/benchmarks/optcarrot/commits.html#chart-1)）、AI（[Rubykon](https://benchmark-driver.github.io/benchmarks/rubykon/commits.html)），以及其它大多数时间都在反复调用某些特定方法的应用中。
+
+虽然 Ruby 3.0 [大幅降低了 JIT 后代码的大小](https://twitter.com/k0kubun/status/1256142302608650244)，对于例如 Rails 这样的负载还没有做好准备。其调用大量不同的方法，在使用 JIT 后会遇到 CPU 指令缓存命中率降低的问题。Ruby 3.1 会在这一问题上进行进一步优化提升，敬请关注。
```
Comment (ecnelises) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:40]: `但对于例如 Rails 这样的负载还没有做好准备`

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 6]
```diff
@@ -0,0 +1,388 @@
...
+目前支持的类和方法有：
+
+- `Mutex#lock`, `Mutex#unlock`, `Mutex#sleep`
+- `ConditionVariable#wait`
+- `Queue#pop`, `SizedQueue#push`
+- `Thread#join`
+- `Kernel#sleep`
+- `Process.wait`
+- `IO#wait`, `IO#read`, `IO#write` and related methods (e.g. `#wait_readable`, `#gets`, `#puts` and so on).
+- `IO#select` is *not supported*.
```
Comment (ecnelises) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:101]:
```suggestion
- `IO#wait`, `IO#read`, `IO#write` 以及其他相关方法（如 `#wait_readable`, `#gets`, `#puts` 等）
- `IO#select` *不被支持*
```

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 7]
```diff
@@ -0,0 +1,388 @@
...
+
+### RBS
+
+RBS 是一门描述 Ruby 类型的语言。
+
+包括 TypeProf 在内的支持 RBS 的类型检查器能更好通过 RBS 定义来理解 Ruby 程序。
+
+你可以写下类和模块的定义：例如类中的方法、实例变量和它们的类型，以及它们如何组合和继承。
+
+RBS 的目的是支持常见的 Ruby 程序的各种模式，并使其可以描述高级类型，包括：组合类型、函数重载、泛型。其也支持对鸭子类型提供接口类型。
```
Comment (ecnelises) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:135]: better as `其也支持对鸭子类型提供接口类型。` -> `也能通过_接口_来支持鸭子类型。`?

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 8]
```diff
@@ -0,0 +1,388 @@
...
+  def initialize : (name: String, age: Integer) -> [String, Integer]
+end
+```
+
+You can run TypeProf by saving the input as "test.rb" and invoke a command called "typeprof test.rb".
+你只需把输入保存为 `test.rb`文件然后调用命令 `typeprof test.rb` 即可生成。
+
+你也可以 [在线尝试 TypeProf](https://mame.github.io/typeprof-playground/#rb=%23+test.rb%0Aclass+User%0A++def+initialize%28name%3A%2C+age%3A%29%0A++++%40name%2C+%40age+%3D+name%2C+age%0A++end%0A++%0A++attr_reader+%3Aname%2C+%3Aage%0Aend%0A%0AUser.new%28name%3A+%22John%22%2C+age%3A+20%29&rbs=)。（其运行在服务器上，如果出现服务中断，我们深表遗憾。）
+
+详见 [文档](https://github.com/ruby/typeprof/blob/master/doc/doc.md) 和 [示例]](https://github.com/ruby/typeprof/blob/master/doc/demo.md)。
```
Comment (ecnelises) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:195]:
```suggestion
详见 [文档](https://github.com/ruby/typeprof/blob/master/doc/doc.md) 和 [示例](https://github.com/ruby/typeprof/blob/master/doc/demo.md)。
```

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 9]
```diff
@@ -0,0 +1,388 @@
...
+  case ["a", 1, "b", "c", 2, "d", "e", "f", 3]
+  in [*pre, String => x, String => y, *post]
+    p pre  #=> ["a", 1]
+    p x    #=> "b"
+    p y    #=> "c"
+    p post #=> [2, "d", "e", "f", 3]
+  end
+  ```
+
+* 加入了无尽函数定义
```
Comment (ecnelises) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:234]:
```suggestion
* 加入了无需 `end` 关键字的函数定义
```

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 10]
```diff
@@ -0,0 +1,388 @@
...
+* 内建了 `Hash#except`
+
+  ``` ruby
+  h = { a: 1, b: 2, c: 3 }
+  p h.except(:a) #=> {:b=>2, :c=>3}
+  ```
+
+* 内存查看以实验性的形式加入
+
+    * 这是一个用来交换内存原始空间的新 C-API。例如将数组和位图在两个扩展库中交换。该扩展库可以互相共享其内存空间的元信息，例如尺寸、格式等。利用这些元信息，扩展库之间可以共享例如多维度的数组。该功能参考了 Python 的 buffer 协议设计。
```
Comment (ecnelises) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:249]: `buffer 协议` -> Use direct `buffer protocol` or `缓冲协议`

https://docs.python.org/zh-cn/3/c-api/buffer.html

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 11]
```diff
@@ -0,0 +1,388 @@
...
+
+    ``` ruby
+    def method_missing(meth, ...)
+      send(:"do_#{ meth }", ...)
+    end
+    ```
+
+* 模式匹配 (`case/in`) 不再是实验性特性。
+* `$SAFE` 特性被彻底移除，现在它就是一个普通的全部常量。
+* bactrace 的顺序在 Ruby 2.5 中被颠倒，现在倒了回来。现在其行为和 2.4 一致，先打印错误信息，再以此打印其调用者。
```
Comment (ecnelises) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:292]: `backtrace`

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 12]
```diff
@@ -0,0 +1,388 @@
...
+  * digest
+  * io-nonblock
+  * io-wait
+  * nkf
+  * pathname
+  * syslog
+  * win32ole
+
+详见 [NEWS](https://github.com/ruby/ruby/blob/{{ release.tag }}/NEWS.md) 或 [提交记录]](https://github.com/ruby/ruby/compare/v2_7_0...{{ release.tag }})
+for more details.
```
Comment (ecnelises) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:354]:
```suggestion
参考 [NEWS](https://github.com/ruby/ruby/blob/{{ release.tag }}/NEWS.md) 或 [提交记录](https://github.com/ruby/ruby/compare/v2_7_0...{{ release.tag }}) 以获取更多信息。
```

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 13]
```diff
@@ -0,0 +1,388 @@
...
+  * StringIO 3.0.0
+  * StringScanner 3.0.0
+  * etc.
+* 标准库以及内置的 gem 不再含有下面的 gems
+  如果要使用相应的特性，请安装对应的 gem
+  * sdbm
+  * webrick
+  * net-telnet
+  * xmlrpc
+* 下面的 gem 被默认内置
```
Comment (ecnelises) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:314]: _Bundled Gems_ are different from _Default Gems_, see https://stdgems.org/#about-the-gemified-standard-library

```suggestion
* 以下 gems 将从默认变为内置
```

**Comment by dsh0416** [comment]
Thanks, @hlcfan @ecnelises . Problems mentioned have been solved. Meanwhile, I don't think we have _italic_ in general Chinese typography. I replace it with **bold** to emphasize the phrases.

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 14]
```diff
@@ -0,0 +1,386 @@
...
+
+> 现在是多核的时代，并发性很重要。有了 Ractor 和异步 Fiber，Ruby 将会成为一门真正的并发语言。—— Matz
+
+### Ractor（实验性）
+
+Ractor 是一个基于 Actor 模型的并发抽象层，提供了无需担心线程安全的并行环境。
+
+你可以创建多个 Ractor，它们能并行运行。Ractor 能保障多线程的线程安全，因为其无法跨线程共享一般对象，需要通过多个 Ractor 之间的通信来交换信息。
+
+为了限制共享对象，Ractor 在 Ruby 语法中引入了多重的限制（如果不使用多个 Ractor 的话，没有这些性质）。
```
Comment (hlcfan) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:52]: "性质" => "限制"

**Review by hlcfan — APPROVED** [review]
Looking good, thanks!

File: zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md [hunk 15]
```diff
@@ -0,0 +1,386 @@
...
+    ```
+
+* 模式匹配 (`case/in`) 不再是实验性特性。
+* `$SAFE` 特性被彻底移除，现在它就是一个普通的全部常量。
+* backtrace 的顺序在 Ruby 2.5 中被颠倒，现在倒了回来。现在其行为和 2.4 一致，先打印错误信息，再以此打印其调用者。
+* 一些标准库被升级
+  * RubyGems 3.2.3
+  * Bundler 2.2.3
+  * IRB 1.2.6
+  * Reline 0.1.5
```
Comment (JuanitoFatas) [inline zh_cn/news/_posts/2020-12-25-ruby-3-0-0-released.md:296]: 有些版本需要改一下

https://github.com/ruby/www.ruby-lang.org/commit/f4a1cdaad8af1cd2b6ae57c62aa61ed36f44b21f#diff-8c5960f133f62c3bf3c753d4bab14c53f6095b1531b88c2a1184a54bab38390f

**Comment by dsh0416** [comment]
Thanks to @hlcfan @JuanitoFatas  for reviewing. The problems mentioned have been fixed.

**Pull request merged**